lisa/tests/test_platform.py

```diff
@@ -0,0 +1,152 @@
+from typing import cast
+from unittest.case import TestCase
+
+from lisa import schema
+from lisa.environment import Environment, Environments, load_environments
+from lisa.platform_ import Platform, load_platform
+from lisa.tests.test_environment import generate_runbook as generate_env_runbook
+from lisa.util import LisaException, constants
+from lisa.util.logger import Logger
+
+
+class MockPlatform(Platform):
+    @classmethod
+    def platform_type(cls) -> str:
+        return constants.PLATFORM_MOCK
+
+    def set_test_config(
+        self, return_prepared: bool = True, deploy_success: bool = True,
+    ) -> None:
+        self.return_prepared = return_prepared
+        self.deploy_success = deploy_success
+
+    def _prepare_environment(self, environment: Environment, log: Logger) -> bool:
+        return self.return_prepared
+
+    def _deploy_environment(self, environment: Environment, log: Logger) -> None:
```

Comment:
Sine this is transforming an environment, doesn't it make sense to return the environment? (Not sure where this is called, but I'm always skeptical of complex functions which return `None` because then we're just relying on side effects.)

Reply:
It created and returned an environment in previous version. But I found the environment contains many information, and it's hard to let every platform have consistent behavior and create an environment correctly. But I added into backlog, after the code is more stable, and implemented more platforms, I will revisit what exactly can be done by deploying. It may be enough to consume information from prepare_env, and return nodes.